app/tray/Account/Account.js

```diff
@@ -301,11 +302,32 @@ class _AccountBody extends React.Component {
         <SignTypedDataRequest
           key={req.handlerId}
           req={req}
+          step={data.step || ''}
           handlerId={req.handlerId}
           accountId={this.props.id}
           signingDelay={signingDelay}
         />
       )
+    } else if (req.type === 'signErc20Permit') {
+      const originName = this.store('main.origins', req.origin, 'name')
+      const chainId = req.typedMessage.data.domain.chainId
+      const chainName = this.store('main.networks.ethereum', chainId, 'name')
+      const { primaryColor: chainColor, icon } = this.store('main.networksMeta.ethereum', chainId)
+      const chainData = { chainName, chainColor, icon }
+      return (
+        <SignPermitRequest
+          key={req.handlerId}
+          {...{
+            req,
+            signingDelay,
+            chainId,
+            originName,
+            chainData
+          }}
+          step={data.step || ''}
```

Comment:
I think whatever passes step should give the correct step, this should never be undefined